odoo/tools/mail.py

```diff
@@ -595,3 +595,30 @@ def formataddr(pair, charset='utf-8'):
                 name=email_addr_escapes_re.sub(r'\\\g<0>', name),
                 addr=address)
     return address
+
+
+def email_domain_extract(email):
+    """Return the domain of the given email."""
+    if not email:
+        return
+
+    email_split = getaddresses([email])
+    if not email_split or not email_split[0]:
+        return
+
+    *_, domain = email_split[0][1].rpartition('@')
+    return domain
+
+
+def encapsulate_email(old_email, new_email):
+    """Change the FROM of the message and use the old one as name.
+    e.g.
+        Old From: "Admin" <admin@gmail.com>
+        New From: notifications@odoo.com
+        Output:   "Admin (admin@gmail.com)" <notifications@odoo.com>
+    """
+    old_email_split = getaddresses([old_email])
+    if not old_email_split or not old_email_split[0]:
+        return old_email
+
+    return '"%s (%s)" <%s>' % (old_email_split[0][0], old_email_split[0][1], new_email)
```

Comment:
Why not
```
return formataddr(
    (("%s (%s)" % (*old_email_split[0])
     new_email)
)
```

Reply:
According to the `getaddresses` doc, it returns "a list of 2-tuples", so `old_email_split[0]` should work as-is.